pkg/agent/proxy/proxier.go

```diff
@@ -613,6 +628,31 @@ func (p *proxier) installServices() {
 	}
 }
 
+func (p *proxier) getMCSExportedServiceInfo(serviceCIDRIPv4 *net.IPNet, endpointIP string, svcPortName k8sproxy.ServicePortName) *openflowtypes.ServiceGroupInfo {
+	var mcsLocalService *openflowtypes.ServiceGroupInfo
+	// When the Endpoint is a local ClusterIP, it means the corresponding local Service is a member of the Multi-cluster Service.
+	if serviceCIDRIPv4 != nil && serviceCIDRIPv4.Contains(net.ParseIP(endpointIP)) {
+		mcsLocalService = &openflowtypes.ServiceGroupInfo{
+			ClusterIP: net.ParseIP(endpointIP),
+		}
+	}
+	var groupID binding.GroupIDType
+	// For any Multi-cluster Service, its name will be a combination with prefix `antrea-mc-` and
+	// exported Service's name. So we need to remove the prefix to look up the exported Service.
+	if mcsLocalService != nil && strings.HasPrefix(svcPortName.Name, mcServiceNamePrefix) {
```

Comment:
Questions:

1. If a local Service is exported after the MC Service is created, I assume the routine will still be triggered to update the MC Service's Group?
2. If the local Service does not have ready endpoints, its ClusterIP will not be added to the MCS Service's Endpoints?

Reply:
1. yes, a new Endpoint will be added to MC Service, so it will trigger and update MC Group.
2. correct, we only include the Service's ClusterIP when it has available Endpoints.